public/app/mu-plugins/moj-auth/moj-auth.php

```diff
@@ -120,6 +120,19 @@ public function handleCallbackRequest(): void
         // The callback has returned an access token.
         if (!is_object($oauth_access_token) || $oauth_access_token->hasExpired()) {
             $this->log('Access token is not valid, or expired.');
+
+            // Update (or create) the JWT to keep track of failed callbacks.
+            $jwt = $this->getJwt() ?: (object)[];
+
+            // Set to 0 for a session cookie.
+            $jwt->cookie_expiry = 0;
+
+            // Set failed_callbacks with a default of 1, or add one to the existing value.
+            $jwt->failed_callbacks = isset($jwt->failed_callbacks) ? ((int) $jwt->failed_callbacks) + 1 : 1;
```

Comment:
Assuming this isn't being dynamically created, If the variable is NULL, `isset()` isn't needed.
```php
$jwt->failed_callbacks = $jwt->failed_callbacks ? ((int) $jwt->failed_callbacks) + 1 : 1;
```

Reply:
I think there will be edge cases where it's not set. It should never be null, if null is passed to `setJwt` then the property `failed_callbacks` is not there.